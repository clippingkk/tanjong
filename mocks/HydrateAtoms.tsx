import { useHydrateAtoms } from 'jotai/utils'

export function HydrateAtoms({ initialValues, children }: any) {
  useHydrateAtoms(initialValues)
  return children
}