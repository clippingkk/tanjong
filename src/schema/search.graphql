query searchQuery($query: String!) {
  search(query: $query) {
    clippings {
      id
      content
    }
  }
}
