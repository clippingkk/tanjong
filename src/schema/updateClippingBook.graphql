mutation updateClippingBookId($cid: Int!, $doubanId: Int!) {
  updateClippingBookId(clippingId: $cid, doubanId: $doubanId) {
    id
    bookID
    title
    content
    createdAt
    pageAt
  }
}
