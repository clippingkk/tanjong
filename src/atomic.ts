import { ProfileQuery } from "./schema/generated"
import { atomWithMMKV } from "./utils/jotai"

export const tokenAtom = atomWithMMKV<string | null>('me:token', null)
export const uidAtom = atomWithMMKV<number | null>('me:uid', null)
export const profileAtom = atomWithMMKV<ProfileQuery['me'] | null>('me:profile', null)