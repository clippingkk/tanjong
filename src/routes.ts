import { Clipping } from "./schema/generated"
import { WenquBook } from "./service/wenqu"

export enum RouteKeys {
  AuthV3 = 'AuthV3',
  SignUp = 'Signup',
  AuthAppleBind = 'AuthAppleBind',
  AuthPhoneOTP = 'AuthPhoneOTP',
  AuthQRCode = 'AuthQRCode',
  TabHome = 'TabHome',
  TabProfile = 'TabProfile',
  Book = 'Book',
  Profile = 'Profile',
  Clipping = 'Clipping',
  BookDetail = 'BookDetail',
  ProfileSettings = 'ProfileSettings',
  ProfileDebug = 'ProfileDebug',
  Payment = 'Payment'
}

export type RouteParamList = {
  empty: object
  Book: {
    book: WenquBook
  }
  Clipping: {
    clipping?: Pick<Clipping, 'id' | 'bookID' | 'content' | 'title'>
    clippingID?: number
  }
  AuthAppleBind: {
    idToken: string
  }
  AuthPhoneOTP: {
    phone: string
    idToken: string
  }
  [RouteKeys.AuthQRCode]: object
  [RouteKeys.Payment]: object
  [RouteKeys.AuthV3]: object
  [RouteKeys.SignUp]: object
  [RouteKeys.BookDetail]: {
    book: WenquBook
  }
  [RouteKeys.ProfileSettings]: object
  [RouteKeys.ProfileDebug]: object

  root: object
}

export type TabRouteParamList = {
  [RouteKeys.TabHome]: object
  [RouteKeys.TabProfile]: {}
}
