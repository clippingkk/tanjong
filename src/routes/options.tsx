import {
  RouteProp,
  Link,
  ParamListBase,
  getFocusedRouteNameFromRoute
} from '@react-navigation/native'
import {
  NativeStackNavigationOptions,
  NativeStackHeaderProps,
  NativeStackNavigationProp
} from '@react-navigation/native-stack'
import {RouteKeys} from '../routes'

export function getRootPageOptions(props: {
  route: RouteProp<ParamListBase, string>
  navigation: NativeStackNavigationProp<ParamListBase, string, undefined>
  theme: ReactNavigation.Theme
}): NativeStackNavigationOptions {
  let headerTitle = ''
  let headerRight: NativeStackHeaderProps['options']['headerRight'] = undefined
  const isRoot = props.route.name === 'root'
  if (!isRoot) {
    return {}
  }

  const routeName =
    (getFocusedRouteNameFromRoute(props.route) as RouteKeys) ??
    RouteKeys.TabHome
  switch (routeName) {
    case RouteKeys.TabHome:
      headerTitle = 'Home'
      // headerRight = () => (
      //   <Link to={{ screen: RouteKeys.Payment }}>
      //     ü§ë
      //   </Link>
      // )
      break
    case RouteKeys.TabSquare:
      headerTitle = 'Square'
      break
    case RouteKeys.TabProfile:
      headerTitle = 'Me'
      headerRight = () => <Link screen={RouteKeys.ProfileSettings}>‚öôÔ∏è</Link>
      break
    default:
      headerTitle = 'ClippingKK'
  }

  return {
    headerTitle,
    headerRight
  }
}
