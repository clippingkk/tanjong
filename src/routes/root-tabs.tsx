import {RouteKeys, RouteParamList, TabRouteParamList} from '../routes'
import {BlurView} from '@react-native-community/blur'
import {useTextColor} from '../hooks/color'
import HomePage from '../pages/home/home.page'
import ProfilePage from '../pages/profile/profile.page'
import SquarePage from '../pages/square'
import {
  HomeIcon,
  UserIcon,
  GlobeAmericasIcon
} from 'react-native-heroicons/outline'
import {useColorScheme} from 'react-native'
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs'
import {createStaticNavigation} from '@react-navigation/native'

const TabStack = createBottomTabNavigator({
  screenOptions: props => {
    const c = useColorScheme()
    const textColor = useTextColor()
    return {
      headerTransparent: true,
      headerShown: true,
      headerTitleStyle: {
        color: c === 'dark' ? '#ffffff' : '#000000'
      },
      tabBarStyle: {
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0
      },
      tabBarBackground: () => (
        <BlurView
          blurType={c === 'dark' ? 'dark' : 'light'}
          blurAmount={10}
          style={{
            position: 'absolute',
            bottom: 0,
            left: 0,
            right: 0,
            top: 0
          }}
        />
      )
    }
  },
  screens: {
    [RouteKeys.TabHome]: {
      screen: HomePage,
      options: {
        title: 'Home',
        tabBarLabel: 'Books',
        tabBarIcon: ({color, size}) => <HomeIcon color={color} size={size} />
      }
    },
    [RouteKeys.TabSquare]: {
      screen: SquarePage,
      options: {
        title: 'Square',
        tabBarLabel: 'Square',
        tabBarIcon: ({color, size}) => (
          <GlobeAmericasIcon color={color} size={size} />
        )
      }
    },
    [RouteKeys.TabProfile]: {
      screen: ProfilePage,
      options: {
        title: 'Profile',
        tabBarLabel: 'Profile',
        tabBarIcon: ({color, size}) => <UserIcon color={color} size={size} />
      }
    }
  }
})

const SVGIconScale = 0.8

// type HomeTabPagesProps = {
// }

// const Navigation = createStaticNavigation(TabStack)

// function RootTabs(props: HomeTabPagesProps) {
//   return <Navigation />
// }

export default TabStack
