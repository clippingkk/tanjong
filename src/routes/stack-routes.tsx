import {
  createNativeStackNavigator,
  NativeStackNavigationOptions
} from '@react-navigation/native-stack'
import BookPage from '../pages/book/book.page'
import DebugPage from '../pages/settings/debug.page'
import SettingsPage from '../pages/settings/settings.page'
import {RouteKeys} from '../routes'
import HomeTabPages from '../pages/home/home.page'
import AuthQRCodePage from '../pages/auth/qrcode.page.mock'
import ClippingPage from '../pages/clipping/clipping'
import AuthV3Page from '../pages/auth/auth.v3'
import AuthApplePhoneBind from '../pages/auth/phone/phone'
import AuthPhoneOTPPage from '../pages/auth/phone/otp'
import PaymentPage from '../pages/payment/payment'
import SignUpEmailPage from '../pages/signup/signup.email'
import SignUpPasswordPage from '../pages/signup/signup.password'
import SignUpOTPPage from '../pages/signup/signup.otp'
import SignUpSetNamePage from '../pages/signup/signup.set.name'
import {getRootPageOptions} from './options'
import RootTabs from './root-tabs'

export const rootStackRoutes = createNativeStackNavigator({
  initialRouteName: 'root',
  screenOptions: ps => getRootPageOptions(ps),
  screens: {
    root: {
      screen: RootTabs,
      // title: 'Home',
      // options: (ps) => getRootPageOptions(ps),
      options: {
        headerShown: false
      }
    },
    [RouteKeys.AuthQRCode]: {
      screen: AuthQRCodePage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.SignUpEmail]: {
      screen: SignUpEmailPage,
      options: {
        headerTransparent: true,
        title: 'Sign Up',
        headerShown: true
      }
    },
    [RouteKeys.SignUpPassword]: {
      screen: SignUpPasswordPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.SignUpOTP]: {
      screen: SignUpOTPPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.SignUpSetName]: {
      screen: SignUpSetNamePage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.AuthV3]: {
      screen: AuthV3Page,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.AuthAppleBind]: {
      screen: AuthApplePhoneBind,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.AuthPhoneOTP]: {
      screen: AuthPhoneOTPPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.BookDetail]: {
      screen: BookPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.ProfileSettings]: {
      screen: SettingsPage,
      options: {
        headerTransparent: true,
        headerTitle: 'settings',
        headerShown: true
      }
    },
    [RouteKeys.ProfileDebug]: {
      screen: DebugPage,
      options: {
        headerTransparent: true,
        headerTitle: 'Debug',
        headerShown: true
      }
    },
    [RouteKeys.Clipping]: {
      screen: ClippingPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    },
    [RouteKeys.Payment]: {
      screen: PaymentPage,
      options: {
        headerTransparent: true,
        headerShown: true
      }
    }
  }
})

export type RootStackRoutes = typeof rootStackRoutes
