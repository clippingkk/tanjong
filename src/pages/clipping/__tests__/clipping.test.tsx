import 'react-native'
import React from 'react'
import { act, render, userEvent, waitFor } from '@testing-library/react-native'
import { RouteKeys, RouteParamList, TabRouteParamList } from '../../../routes'
import { NativeBaseProvider, Text } from 'native-base'
import { MockedProvider } from "@apollo/client/testing"
import { NavigationContainer } from '@react-navigation/native'
import { GluestackUIProvider } from '@gluestack-ui/themed'
import { config } from '@gluestack-ui/config'
import { uidAtom } from '../../../atomic'
import { FetchClippingAiSummaryDocument, FetchClippingDocument, } from '../../../schema/generated'
import { HydrateAtoms } from '../../../../mocks/HydrateAtoms'
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import ClippingPage from '../clipping'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { mockedBook } from '../../../../mocks/data'

jest.mock('../../../service/wenqu', () => {
  return {
    __esModule: true,
    wenquRequest: jest.fn(() => {
      return {
        count: 1,
        books: [{
          ...mockedBook,
          doubanId: '27036197'
        }]
      }
    })
  }
})

test('book page renders with permission block', async () => {
  const st = createNativeStackNavigator<RouteParamList>()
  const qc = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: Infinity,
        retry: false,
      }
    }
  })
  const canvas = render(
    <MockedProvider mocks={[
      {
        delay: 30,
        request: {
          query: FetchClippingAiSummaryDocument,
          variables: {
            id: 70710,
          }
        },
        result: {
          data: {
            clipping: {
              "id": 70710,
              "aiSummary": "the ai summary should be here",
              "__typename": "Clipping"
            }
          }
        }
      },
      {
        delay: 30,
        request: {
          query: FetchClippingDocument,
          variables: {
            id: 3333,
          }
        },
        result: {
          data: {
            clipping: {
              "id": 70710,
              "bookID": "27036197",
              "title": "Odd Arne Westad - The Cold War_ A World History-Basic Books",
              "content": "The desperation created by total war in Europe and the fear that it would spread to much of the rest of the globe was in the minds of all those who experienced it, regardless of where they experienced it.",
              "createdAt": "2023-12-01T09:42:59Z",
              "pageAt": "#419-420",
              "visible": true,
              "prevClipping": {
                "userClippingID": 70709,
                "bookClippingID": 70709,
                "__typename": "siblingClippingObject"
              },
              "nextClipping": {
                "userClippingID": 0,
                "bookClippingID": 0,
                "__typename": "siblingClippingObject"
              },
              "reactionData": {
                "count": 0,
                "symbolCounts": [],
                "__typename": "ReactionData"
              },
              "creator": {
                "id": 1,
                "name": "AnnatarHe",
                "avatar": "clippingkk/avatar/user-BpLnfgDsc3WD9F3qNfHK6a95jjJkwzDk.jpg",
                "domain": "annatar.he",
                "premiumEndAt": "2023-12-27T14:57:15Z",
                "__typename": "User"
              },
              "comments": [],
              "__typename": "Clipping"
            }
          }
        }
      },
    ]}>
      <NativeBaseProvider initialWindowMetrics={{
        frame: { x: 0, y: 0, width: 0, height: 0 },
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
      }}>
        <QueryClientProvider client={qc}>
          <HydrateAtoms initialValues={[[uidAtom, 1]]}>
            <GluestackUIProvider config={config}>
              <NavigationContainer>
                <st.Navigator>
                  <st.Screen
                    component={ClippingPage}
                    name={RouteKeys.Clipping}
                    initialParams={{
                      clippingID: 3333
                    }}
                  />
                </st.Navigator>
              </NavigationContainer>
            </GluestackUIProvider>
          </HydrateAtoms>
        </QueryClientProvider>
      </NativeBaseProvider>
    </MockedProvider>
  )

  await act(() => { })
  await waitFor(() =>
    expect(canvas.getByText('The desperation created by total war in Europe and the fear that it would spread to much of the rest of the globe was in the minds of all those who experienced it, regardless of where they experienced it.')).toBeOnTheScreen()
  )

  expect(canvas).toMatchSnapshot()

  await userEvent.press(
    canvas.getByRole('button', { name: 'âœ¨ AI Description' })
  )

  await waitFor(() =>
    expect(canvas.getByText('the ai summary should be here')).toBeOnTheScreen()
  )
})
