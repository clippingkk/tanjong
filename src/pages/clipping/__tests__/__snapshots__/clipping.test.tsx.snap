// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`book page renders with permission block 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            collapsable={false}
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <RCTScrollView
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              refreshControl={
                <RefreshControlMock
                  onRefresh={[Function]}
                  refreshing={false}
                />
              }
              style={
                [
                  undefined,
                  {},
                  {},
                  {},
                  {
                    "backgroundColor": "#f1f5f9",
                    "height": "100%",
                  },
                  undefined,
                ]
              }
              sx={
                {
                  "_dark": {
                    "backgroundColor": "$blueGray700",
                  },
                }
              }
            >
              <RCTRefreshControl />
              <View>
                <View
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  style={
                    [
                      undefined,
                      {},
                      {},
                      {},
                      {
                        "height": "100%",
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <RCTSafeAreaView>
                    <View
                      data-style=""
                      dataSet={
                        {
                          "style": "",
                        }
                      }
                      style={
                        [
                          undefined,
                          {},
                          {},
                          {},
                          {
                            "paddingTop": 32,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        data-style=""
                        dataSet={
                          {
                            "style": "",
                          }
                        }
                        selectable={true}
                        size="md"
                        style={
                          [
                            undefined,
                            {},
                            {
                              "color": "#525252",
                              "fontFamily": undefined,
                              "fontStyle": "normal",
                              "fontWeight": "400",
                              "letterSpacing": 0,
                            },
                            {},
                            {
                              "fontSize": 16,
                              "lineHeight": 22,
                            },
                            {
                              "fontSize": 16,
                              "lineHeight": 22,
                            },
                            {
                              "fontFamily": "LXGWWenKai-Regular",
                              "fontSize": 18,
                            },
                            undefined,
                          ]
                        }
                      >
                        The desperation created by total war in Europe and the fear that it would spread to much of the rest of the globe was in the minds of all those who experienced it, regardless of where they experienced it.
                      </Text>
                    </View>
                  </RCTSafeAreaView>
                  <View
                    dataSet={{}}
                    style={
                      [
                        {},
                        {
                          "marginTop": 10,
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      colorScheme="primary"
                      dataSet={{}}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#0891b2",
                          "borderRadius": 4,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "paddingBottom": 10,
                          "paddingLeft": 12,
                          "paddingRight": 12,
                          "paddingTop": 10,
                        }
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                        test={true}
                      >
                        <View
                          dataSet={{}}
                          style={{}}
                        >
                          <Text
                            dataSet={{}}
                            style={
                              {
                                "backgroundColor": undefined,
                                "color": "#fafafa",
                                "fontFamily": undefined,
                                "fontSize": 14,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                                "letterSpacing": 0,
                                "lineHeight": 21,
                                "textDecorationLine": undefined,
                              }
                            }
                          >
                            ✨ AI Description
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    data-style=""
                    dataSet={
                      {
                        "style": "",
                      }
                    }
                    orientation="horizontal"
                    style={
                      [
                        undefined,
                        {},
                        {
                          "backgroundColor": "#DBDBDB",
                        },
                        {},
                        {
                          "height": 1,
                          "width": "100%",
                        },
                        {
                          "height": 1,
                          "width": "100%",
                        },
                        {
                          "marginTop": 16,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </RCTScrollView>
            <RCTSafeAreaView
              collapsable={false}
              onLayout={[Function]}
              pointerEvents="none"
              style={
                {
                  "backgroundColor": "transparent",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 1,
                }
              }
            >
              <View />
            </RCTSafeAreaView>
          </View>
        </View>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleVisible={true}
          backgroundColor="transparent"
          blurEffect="light"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          hideShadow={true}
          largeTitleHideShadow={false}
          title="Odd Arne Westad - The Cold War_ A World History-Basic Books"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={true}
        >
          <RNSScreenStackHeaderSubview
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            type="right"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              action="primary"
              collapsable={false}
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              size="xs"
              states={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              style={
                [
                  undefined,
                  {},
                  {
                    "alignItems": "center",
                    "backgroundColor": "#0077E6",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  {},
                  {
                    "backgroundColor": "#0077E6",
                    "borderColor": "#4AA9FF",
                  },
                  {
                    "paddingHorizontal": 0,
                  },
                  {
                    "height": 32,
                    "paddingHorizontal": 14,
                  },
                  {
                    "backgroundColor": "transparent",
                    "paddingHorizontal": 0,
                  },
                  undefined,
                ]
              }
              variant="link"
            >
              <Text
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                size="md"
                style={
                  [
                    undefined,
                    {},
                    {
                      "color": "#525252",
                      "fontFamily": undefined,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {},
                    {
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    undefined,
                  ]
                }
              >
                 🌐 
              </Text>
            </View>
          </RNSScreenStackHeaderSubview>
        </RNSScreenStackHeaderConfig>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
